### PR TITLE
Handling compass being nan from IMUSensor

### PR DESCRIPTION
**Pull request by emailweixu**
Also do not use docker to run carla if already in docker container.
And make sure vehicles not reset to a location colliding with other vehicles.

### CODE REVIEW

File: alf/environments/suite_carla.py [hunk 1]
```diff
@@ -334,11 +335,15 @@ def _IMU_callback(weak_self, sensor_data):
         self = weak_self()
         if not self:
             return
+        if not math.isnan(sensor_data.compass):
+            self._compass = sensor_data.compass
```
Comment (le-horizon) [inline alf/environments/suite_carla.py:339]: maybe initialize self._compass to some default value in __init__()?

**Review by le-horizon — APPROVED** [review]
LG.  one nit.

File: alf/environments/suite_carla.py (hunk 1 above)
Comment (emailweixu) [reply alf/environments/suite_carla.py:339]: "self._compass  = 0" is there already

Comment (le-horizon) [reply alf/environments/suite_carla.py:339]: Ah, missed that..  LG then.

File: alf/utils/common.py [hunk 2]
```diff
@@ -856,3 +856,26 @@ def flattened_size(spec):
...
+
+def is_inside_docker_container():
+    """Return whether the current process is running inside a docker container.
+
+    See discussions at `<https://stackoverflow.com/questions/23513045/how-to-check-if-a-process-is-running-inside-docker-container>`_
+    """
+    return os.path.exists("/.dockerenv")
+
+
+def check_numerics(nested):
```
Comment (hnyu) [inline alf/utils/common.py:869]: There are several other places in the current code that have a simpler check_numerics. Maybe replace them with this function?

Comment (emailweixu) [reply alf/utils/common.py:869]: This is good for nested one. The other places are not checking nested.

**Review by hnyu — APPROVED** [review]

**Pull request merged**